### DISCUSSION TITLE
Will this continue as source of the Faker library? #1

### DISCUSSION THREAD

**Discussion by mmahalwy** [faker-js/faker#52] [General]
@Geertvdc asked in https://github.com/withshepherd/faker.js/issues/1#issue-1094203363:

```
I noticed this morning that the original faker repo and npm package were all removed or emptied (https://github.com/Marak/Faker.js). Looking for a fork or alternative I bumped into this.

Will you continue building / supporting faker.js? Do you need any help? Could you elaborate on the plans? Did you know Marak?

I think Faker.js by Marak was a well used library and it would be a shame if it would go away. I would love helping out in getting this back up if you need any help.
```

Continuing conversation here

**Comment by prisis** [comment]
I will try to port old PRs from web archive to this repo

**Comment by damienwebdev** [comment]
@prisis just to ensure that we respect the original contributor of the PR, can you ref the PR (or the author here) so that we can ping them and have them submit it themselves if they want to?

**Comment by prisis** [comment]
Yeah can do it

**Comment by G1itcher** [comment]
This will probably be an unpopular opinion, but it seems a little disrespectful to do this?

I believe Marak took his project down specifically to protest exploitation of open source by large companies after losing almost everything in a fire? To nullify his protest like this seems a little wrong to me.

**Comment by zachariahtimothy** [comment]
> This will probably be an unpopular opinion, but it seems a little disrespectful to do this?
> 
> I believe Marak took his project down specifically to protest exploitation of open source by large companies after losing almost everything in a fire? To nullify his protest like this seems a little wrong to me.

Is certainly an interesting take. Was it respectful of him to cost our team a couple hours of collective productivity by releasing a new version with no code and leaving us scratching our heads why the build was broken? 
It was his prerogative to do that but this is the nature of open source. This library is necessary and utilized by many. Perhaps he should have been appropriately compensated, but to say nobody is allowed to maintain a fake data library because the original author was unsponsored does not make sense to me.

**Reply by StanleyMasinde** [reply]
You sound so entitled. I had stuff break too but that doesn't mean I blame others for my problem. 

Respectfully it was his code and he decides what to do with it.

I would also like to make it clear that I support this but please let us not forget what Marak was fighting against. 

Thank you

**Reply by damienwebdev** [reply]
While I appreciate the discussion on the repo, let's keep it civil. 

>You sound so entitled. 

Is a judgment on the other's discussion. 

Instead, state your point, be precise, and focus on how we can move `@faker-js/faker` forward.

**Comment by G1itcher** [comment]
I acknowledge that it's a complicated subject that can't be easily slotted on either side of anyone's morality. Just thought it was worth expressing.

**Comment by zachariahtimothy** [comment]
Best thing we can do to avoid this type of thing in the future is to our employers to do more sponsorship! Libraries like this save many many hours of development so why not give even a couple hundred bucks?

**Comment by prisis** [comment]
Hey @MilosPaunovic and @gustawdaniel-acaisoft do you like to work with us on faker.js v6?

**Comment by MilosPaunovic** [comment]
@prisis Hmm... You mean to discard the `community-faker` altogether?

I'm up for it, as I think it will benefit the open source community more than to have multiple versions of, basically, the same thing. Did you have anything particular on your mind?

**Comment by prisis** [comment]
@MilosPaunovic Yes :), we would give to maintainer rights if the other team member are fine with it and if you want it :) Nothing particular in mind just to work together :)

**Reply by MilosPaunovic** [reply]
Agreed; Daniel and I aren't quite a team, per se, he just submitted PR for improvement.

So, all in all, I do agree that me joining you guys on this repo would be the best action, community wise. When you add me as member of the organization, I'm gonna update `README.md` on `community-faker` to point here.

What are we gonna do regarding a package name and publishing on `npm`?

**Reply by damienwebdev** [reply]
@MilosPaunovic we've already decided on[ `@faker-js/faker` and published all previous versions on npm backward compatibly](https://github.com/faker-js/faker). See https://github.com/faker-js/faker/issues/5 for the method used to achieve this.

**Reply by MilosPaunovic** [reply]
Yeah, that sounds good!

**Reply by MilosPaunovic** [reply]
Pointing the https://github.com/MilosPaunovic/community-faker#readme to here!

**Comment by Geertvdc** [comment]
> This will probably be an unpopular opinion, but it seems a little disrespectful to do this?
> 
> I believe Marak took his project down specifically to protest exploitation of open source by large companies after losing almost everything in a fire? To nullify his protest like this seems a little wrong to me.
> 

Do you have a better Idea what to do? Lots of us (not all working for large orgs) depend on this library (with a MIT license)

There are different ways to protest but just deleting something that others also helped in creating (looking at the amount of contributors) doesn't seem the right way. I fully agree that people just continue it's work while honoring the work that Mark did. He basically left us no other choice

**Comment by Icemic** [comment]
*You have no better idea, so we have to choose it.* Reasonable, and however, it implies kind of *necessary evil*. So it's not something that shouldn't be discussed.

Maybe, maybe you're right, he may have many other choices to protest...and it's entirely possible he didn't as well. Just imagine the helplessness and anger of a newly homeless individual in the face of an oligarch who owns his protential property.

Back to topic, is it really no better way? Make a API-compatibility new one with another name, instead of RESTORE it as-is infavor of his decision and protest, and if you want, you can attach the whole story of this incident , which it is much better for Marak, though it is not better enough to solve your actual problem asap.

Sorry for my unpopular opinion.

**Reply by Shinigami92** [reply]
What we are currently doing with faker is IMO just the beginning. We have plans to migrate the whole code to TypeScript and improve docs and add new stuff...
So we just use the source as a starting point

Also we could think about to pipe every sponsored penny to this repo to Marak :thinking:

**Reply by Icemic** [reply]
That's great 👍

**Reply by damienwebdev** [reply]
@Icemic see the discussion here https://github.com/faker-js/faker/discussions/56#discussioncomment-1940760 I have agreement with Marak in intent, but not implementation. 

FOSS is extremely difficult when done as a full-time venture and we (the FOSS community) need a funding strategy that returns value provided back to the maintainers.

As such, we've decided to keep his name, and solely his name, in the FUNDING.yml for that very reason.

**Reply by Icemic** [reply]
> @Icemic see the discussion here [#56 (comment)](https://github.com/faker-js/faker/discussions/56#discussioncomment-1940760) I have agreement with Marak in intent, but not implementation.
> 
> FOSS is extremely difficult when done as a full-time venture and we (the FOSS community) need a funding strategy that returns value provided back to the maintainers.
> 
> As such, we've decided to keep his name, and solely his name, in the FUNDING.yml for that very reason.

Thank you for replyng. I... i am just have some anger to the words blaming the original author which I have listened for the whole day.

It's glad to know your thoughts and I am respect and admire to your efforts as well as the brave decision about funding. May be a good idea to pin a issue explaining these thoughts which could help avoiding some controversy.

**Reply by damienwebdev** [reply]
![Screenshot 2022-01-10 124754](https://user-images.githubusercontent.com/9029654/148813810-3bd17c24-0638-4c5b-9578-679f47ff4410.png)

**Comment by AllanOricil** [comment]
I dont think this would be what marak wanted. But I see he is a sponsor of this project now.

**Reply by damienwebdev** [reply]
@AllanOricil to make sure that no one misinterprets what you wrote. Marak does NOT sponsor this repo, but as described here: https://github.com/faker-js/faker/discussions/56 we have left him as the target of sponsorships of this project.

**Reply by bcheidemann** [reply]
There's been a lot of speculation around what Marak may or may not have wanted. Fortunately, he told everyone exactly what he wanted in November 2020. In his own words:

"Pay Me or Fork This" ([http://web.archive.org/web/20210704022108/https://github.com/Marak/faker.js/issues/1046](http://web.archive.org/web/20210704022108/https://github.com/Marak/faker.js/issues/1046))

The project has now been forked and is being maintained by the community, as per Maraks wishes.

**Reply by AllanOricil** [reply]
Clearly you are misinterpreting what he "really" wanted when he said this phrase. He was clearly trying to put pressure on big companies that use his work without supporting him. He wanted to get attention and get paid for that. He deleted it to show he had the power and guts to pressure those companies. Now forking his work It is really not helping him with that. It is also unpolite and unprofessional. Any professional that use this should be ashamed. Specially the ones who turned in the "official" faker repo.

**Reply by Shinigami92** [reply]
If we would not do it, someone else would, there are already some other forks, but we try to contact them and work together here together.

Beside that, seems everyone has its own interpretation 🤷

**Reply by damienwebdev** [reply]
@AllanOricil I'm the one who:

1. Had a fork of `Marak/faker.js` at the most recent commit which became the basis of this repository.
2. Depended on the library so much that I opted to organize the effort with a group of people to create this org/repo.
3. Personally did the work to republish the packages myself under `@faker-js/faker`.

**Did I do this to invalidate Marak's protest?** Absolutely not. I did it because I need Faker for my project [Daffodil](https://github.com/graycoreio/daffodil) to be stable. 

**Should I have contributed to Faker fiscally?** Absolutely. But, guess what? What capital I have earned from my projects I have fully invested directly back into my Open Source projects because I see the value that they provide to the ecosystem. 

**Did Marak hurt big corporations by doing what he did?** Perhaps, but to what extent? I suspect very little because most of them have private registries they control which cache from NPM. That said, what I believe he did is start a conversation that we all need to be having.

**So, who did he hurt?** Me. My tiny open source project where any time I spend un****ing something is time spent that I can't spend making my project better. 


This all said, [see my comment](https://github.com/faker-js/faker/discussions/56#discussioncomment-1940760). This is originally his work, and I fully intend to respect that as best as I can. 

To your point, am I ashamed? Absolutely not.

**Reply by JessicaSachs** [reply]
@AllanOricil I'll respond to your comment, but I'm not going to engage any further. I need to get back to drafting the official statement we're making.

**Reading minds**

> Clearly you are misinterpreting what he "really" wanted when he said this phrase. He was clearly trying to put pressure on big companies that use his work without supporting him. He wanted to get attention and get paid for that. He deleted it to show he had the power and guts to pressure those companies. Now forking his work It is really not helping him with that. 

First, it is not my job to interpret or read the minds of fully capable adults. I do not engage in "reading between the lines" and I encourage you not to either. You're drawing your own conclusions and I don't particularly trust second/third hand interpretations.

I personally tried to talk to him today and was told by his friends that he was in no condition to speak to anyone. Again, not my business.

**Pride**

> Now forking his work It is really not helping him with that. It is also unpolite and unprofessional. Any professional that use this should be ashamed. Specially the ones who turned in the "official" faker repo.

### I am not ashamed, quite the opposite, I am incredibly proud of what our group of 6 strangers has accomplished in the last 3 days.

This is an impossible position and we've done the best we can by speaking with external parties about how to navigate where we're at.  Especially as a group of people who have never even spoken with each other.

**Professionalism**

Please educate yourself on the history of Open Source Software and the Free Software Movement (they're not the same thing) and learn your history before making character judgements on professionals who work in Open Source Software full-time. We are doing things in line with the philosophies of OSS, so calling us unprofessional doesn't really follow.

The GNU Foundation, despite all of its social problems, has done a [LOT of writing on the philosophies of why software should be free](https://www.gnu.org/philosophy/open-source-misses-the-point.en.html), and among other things, it's because OSS protects individuals and IDEAS from the risk of a single person or company. Start there and read 2-3 articles.

**Lastly, if you would like to act in protest of our decision to support the OSS community we are a part of, please feel free to delete your own repositories.**

I personally think that would be a fitting way to show support.

**Comment by JessicaSachs** [comment]
### This repo will continue as the No. 1 source of the Faker library

We're excited by the outpouring of support from the community and will welcome anyone who loves the project like we do. Please join us in [Discord](https://discord.com/invite/4qDjAmDj4P) if you're interested in contributing.

#### Staying in the loop
Everything is communicated via the official Twitter account https://twitter.com/faker_js. As stated in the [pinned tweet](https://twitter.com/_jessicasachs/status/1481316029636784128?s=20) I will be writing up a statement shortly from the team and will also post a link to this statement shortly.

Having a Twitter account for a library may seem like overkill, but because of how much chaos there is, I think over-communicating is the best plan.
